### PR TITLE
Plans 2023: update copy for sharing feature

### PR DESCRIPTION
**Pull request by aneeshd16**
Fixes Automattic/martech#1501

## Proposed Changes

This PR is based on top of https://github.com/Automattic/wp-calypso/pull/74561.

* The current comparison grid contains 2 separate features: `Limited shares in social media` and `Unlimited shares in social media`. This change replaces these 2 features with a new feature, "Shares on social media" with plan-specific copy.

image
image

_**Note: String Freeze label will be added after the review is complete.**_

## Testing Instructions

* Go to `/start/plans` and scroll down to the comparison grid.
* Confirm that under the "Growth and monetization tools" heading, the new feature is visible and matches the screenshot for both desktop and mobile.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/lib/plans/features-list.tsx [hunk 1]
```diff
@@ -1925,6 +1931,30 @@ export const FEATURES_LIST: FeatureList = {
...
+				return i18n.translate( '%d shares per month', { args: [ 30 ] } );
+			}
+			if (
+				isPremiumPlan( planSlug ) ||
+				isBusinessPlan( planSlug ) ||
+				isEcommercePlan( planSlug )
+			) {
+				return i18n.translate( 'Unlimited shares' );
+			}
+			return '';
```
Comment (aneeshd16) [inline client/lib/plans/features-list.tsx:2084]: Not a fan of bringing in plan functions in the feature list, but I could not find an alternate way to do this in the current plans framework. Open to suggestions!

Comment (southp) [reply client/lib/plans/features-list.tsx:2084]: I agree. Until the day our data definition allows us to write only by some simple slugs, I think this is the best we can do. 🤷🏼

**Review by southp — APPROVED** [review]
Works as described and LGTM :)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7889493

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7889493

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by aneeshd16** [comment]
Rebased.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7889493).

**Pull request merged**